joi-validations/topics.validations.js

```diff
@@ -5,6 +5,16 @@ const TopicValidation = {
     query: {
       name: string.required()
     }
+  },
+  checkName: {
+    query: {
+      name: string.required()
+    }
+  },
+  create: {
+    body: {
+      name: string.required()
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `checkName` and `create` validation objects seem to be identical in their structure. They both require a `name` field. However, the `checkName` object is validating a query parameter while the `create` object is validating a body parameter. This could potentially lead to confusion or misuse. Consider renaming these validation objects to reflect their usage more accurately, such as `checkNameQuery` and `createBody`.

```diff
-  checkName: {
-    query: {
-      name: string.required()
-    }
-  },
-  create: {
-    body: {
-      name: string.required()
-    }
-  }
+  checkNameQuery: {
+    query: {
+      name: string.required()
+    }
+  },
+  createBody: {
+    body: {
+      name: string.required()
+    }
+  }
```
This change will make it clear what each validation object is intended for, improving maintainability and reducing potential for errors.


<!-- This is an auto-generated comment by OSS CodeRabbit -->